src/feature.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  EvaluationApi,
+  EvaluationBatchRequest,
+  EvaluationBatchRequestFlagTagsOperatorEnum,
+  EvaluationEntity,
+  createConfiguration,
+  ServerConfiguration,
+} from "flagr-client";
+
+export interface FlagCallbacks {
+  [key: string]: (...args: any) => Promise<any>;
+}
+
+export interface Variant {
+  key: string | null;
+  attachment: any | null;
+}
+
+const nullVariant: Variant = {
+  key: null,
+  attachment: null,
+};
+
+export interface Config {
+  flagrUrl: string;
+  tags?: string[];
+  tagOperator?: EvaluationBatchRequestFlagTagsOperatorEnum;
+}
+
+export function createFeature(config: Config) {
+  return new Feature(config);
+}
+
+export class Feature {
+  private _context: Object = {};
+  private _evaluationResults: Map<string, Variant> = new Map();
+  private _api: EvaluationApi;
```

Comment:
**nitpick:** undies aren't necessary with the `private` modifier. Alternatively you could [use #s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields)‏